### PR TITLE
[ fix #3604 ] add guide for working around with-termination issues to changelog

### LINKED ISSUE #3604: With-abstraction fools the productivity checker
Following on from [Thorsten's post to the mailing list](https://lists.chalmers.se/pipermail/agda/2019/010902.html), in the following code, Agda mistakenly accepts `f` as productive. We can use this to make the type checker loop on `test`.

```agda
data Maybe (A : Set) : Set where
  just : A → Maybe A
  nothing : Maybe A

data Is-nothing {A} : Maybe A → Set where
  nothing : Is-nothing nothing

record ℕ∞ : Set where
  coinductive
  field
    prd∞ : Maybe ℕ∞

open ℕ∞

f :  ℕ∞ → ℕ∞
prd∞ (f m) with prd∞ m
prd∞ (f m) | just pm with f pm
prd∞ (f m) | just pm | x = prd∞ x
prd∞ (f m) | nothing = nothing

inf∞ : ℕ∞
prd∞ inf∞ = just inf∞

test : Is-nothing (prd∞ (f inf∞))
test = nothing
```

If this
```
prd∞ (f m) | just pm with f pm
prd∞ (f m) | just pm | x = prd∞ x
```
is replaced by this
```
prd∞ (f m) | just pm = prd∞ (f pm)
```
the productivity checker will notice that this definition is not productive. But it believes that the version of `f` with the `with` is productive, and upon trying to run it fails to produce a result.

Agda version 2.5.4.2

### PR DESCRIPTION
**Pull request by UlfNorell**

### CODE REVIEW

**Review by gallais — APPROVED** [review]
Should this also go in the user manual?

**Comment by UlfNorell** [comment]
Yes, I'll add a section to the with-abstraction docs.

File: CHANGELOG.md [hunk 1]
```diff
@@ -292,6 +292,77 @@ Language
...
+
+  ```agda
+  fixed : D → Nat
+  fixed [ zero  ] = zero
+  fixed [ suc n ] with fixed [ n ] | some-stuff
+  ... | rec | _ = rec
+  ```
+
+  If the function takes more arguments you might need to abstract over a
+  partial application to just the structually recursive argument. For instance,
```
Comment (jespercockx) [inline CHANGELOG.md]: typo: structually

**Review by jespercockx — COMMENTED** [review]
LGTM (apart from one typo)

File: CHANGELOG.md (hunk 1 above)
Comment (UlfNorell) [reply CHANGELOG.md]: Isn't that how you actrually spell it?

**Pull request merged**